### ISSUE TITLE
Possible `truthy-bool` false positive with `:=`

### ISSUE THREAD

**Issue by sobolevn** [python/mypy#15685]
Consider this code with `truthy-bool`:

```python
class A:
    def method(self) -> bool:
        return True  # whatever


if (a := A()) and a.method():
    print(a)
```

It will output:

```
» mypy ex.py --enable-error-code=truthy-bool
ex.py:6: error: "ex.a" has type "A" which does not implement __bool__ or __len__ so it could always be true in boolean context  [truthy-bool]
ex.py:6: error: "ex.A" returns "A" which does not implement __bool__ or __len__ so it could always be true in boolean context  [truthy-bool]
```

Two questions:
1. Why two errors?
2. Maybe we should skip this rule completely for `(:=) and ...`, because `a` is not really the main thing we check here?

CC @ikonst

**Comment by AlexWaygood** [comment]
I think it would _probably_ be better to emit one error rather than two here. But I don't think I agree that it's a false positive. The first side of the left operand will always evaluate to `True`, so you could write the snippet more simply as:

```py
class A:
    def method(self) -> bool:
        return True  # whatever


if (a := A()).method():
    print(a)
```

**Comment by ikonst** [comment]
Agree with (1), need to look into it.

Don't see compelling evidence to special-case (2). This code uses `and` creatively as a statement separator (since lhs is always true), and I'm not convinced this is idiomatic or common (just browsed over examples in PEP-572 but I might've missed something).

**Comment by huix11** [comment]
4
